### ISSUE TITLE
Project Name with Extra Double Quote Causes pyproject.toml Error

### ISSUE THREAD

**Issue by NADOOITChristophBa** [beeware/briefcase#2071]
### Describe the bug

When generating a new application with Briefcase, if the project name contains an extra double quote, it results in a syntax error in the pyproject.toml file. This error prevents the application from running.

### Steps to reproduce

Run the briefcase new command.
Enter a project name with an extra double quote, e.g., NADOO Websurfer".
Complete the rest of the prompts to generate the application.
Attempt to run the application with briefcase dev.
See the error in the terminal output.

### Expected behavior

The briefcase new command should either stop with a warning or remove the symbol

### Environment

Operating System: 14.5 (23F79) (macOS)
Python version: 3.12.7
Software versions:
Briefcase: 0.3.16
Toga: 0.4.1

### Logs

```

```

**Comment by freakboy3742** [comment]
Thanks for the report - I thought we had found all of these, but obviously not.

**Comment by dev-adas** [comment]
I'm on it!

**Comment by dev-adas** [comment]
Running `briefcase dev` does not reproduce the error, however `briefcase run` does.

Log error:
```python
Using app template: https://github.com/beeware/briefcase-macOS-app-template.git, branch v0.3.23
Cloning template 'https://github.com/beeware/briefcase-macOS-app-template.git'...
Using existing template (sha f63353430d7c94e2efd4164c9dcd526ba5d03b5e, updated Thu Apr 17 13:23:14 2025)
  File "/var/folders/_p/5_kmj5r5145_n0410jzn6_xh0000gn/T/tmpcb7a8dh2.py", line 12
    INFO_PATH = Path("badFormalName".app/Contents/Info.plist")
                                                         ^
SyntaxError: unterminated string literal (detected at line 12)
Stopping generation because post_gen_project hook script didn't exit successfully
```

One proposed solution was to make the "fix" with `{{ cookiecutter.formal_name|replace('"', '\\"') }}`  [on line 12 of the macOS post gen template script](https://github.com/beeware/briefcase-macOS-app-template/blob/9d8494a5a365c74f03b075aa8f76c48494516b0b/hooks/post_gen_project.py#L12)

However, fixing this caused cascading issues (invalid strings) with the generation of the `briefcase.toml` corresponding to the user's application build.

See example for formal_name as `badFormalName"`
```python
# Generated using Python 3.13.3
[briefcase]
# This is the start of the framework-based support package era.
target_version = "0.3.20"

[paths]
app_path = "badFormalName".app/Contents/Resources/app"
app_packages_path = "badFormalName".app/Contents/Resources/app_packages"
info_plist_path = "badFormalName".app/Contents/Info.plist"
entitlements_path = "Entitlements.plist"
support_path = "badFormalName".app/Contents/Frameworks"
runtime_path = "Python.xcframework/macos-arm64_x86_64/Python.framework"
support_revision = 6
stub_binary_revision = 11
cleanup_paths = [
    "badFormalName".app/Contents/Frameworks/Python.framework/**/Headers",
    "badFormalName".app/Contents/Frameworks/Python.framework/**/include",
    "badFormalName".app/Contents/Frameworks/Python.framework/**/config-*-darwin",
    "badFormalName".app/Contents/Frameworks/Python.framework/**/pkg-config",
]
icon = "badFormalName".app/Contents/Resources/badFormalName.icns"
```

**Comment by freakboy3742** [comment]
> Running `briefcase dev` does not reproduce the error, however `briefcase run` does.

Yes - that's because `briefcase dev` effectively doesn't use the formal name at all - it's really just invoking `python -m app_name`. 

> However, fixing this caused cascading issues (invalid strings) with the generation of the `briefcase.toml` corresponding to the user's application build.

That's not entirely surprising - effectively anywhere that a value is used in a templated file, it needs to be escaped appropriately.

**Comment by kattni** [comment]
@freakboy3742 This looks like it can be closed.

**Comment by freakboy3742** [comment]
Yup - looks like you're right.
